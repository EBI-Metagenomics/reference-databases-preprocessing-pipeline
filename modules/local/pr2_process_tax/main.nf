
process PR2_PROCESS_TAX {

    label 'light'

    input:
    publishDir "${params.outdir}/${label}/${version}/", mode: 'copy'

    path tax
    path tax_header
    val version
    val label

    output:
    path("PR2-tax.txt"), emit: tax

    script:
    """
    sed 's/;/\t/g' $tax > tab-tax.txt
    pr2_process_tax.py -t tab-tax.txt -o processed_tax.txt
    cat $tax_header processed_tax.txt > ${label}-tax.txt
    """

}