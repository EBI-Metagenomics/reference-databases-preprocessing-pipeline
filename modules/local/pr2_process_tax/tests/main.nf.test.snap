{
    "tax": {
        "content": [
            [
                "PR2-tax.txt:md5,09c0e41a85d3bb2abbc66ab125a299c0"
            ]
        ],
        "timestamp": "2023-12-12T14:46:24.504901874"
    }
}